### PR TITLE
Feat: Add contributing guide

### PR DESCRIPTION
**Pull request by kodumbeats**
This PR adds a CONTRIBUTING.md guide which details the PR process and some guidance for creating new adapters.

### CODE REVIEW

**Review by eldadfux — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+- **feat** - is a new feature
+- **doc** - documentation only changes
+- **cicd** - changes related to CI/CD system
+- **fix** - a bug fix
+- **refactor** - code change that neither fixes a bug nor adds a feature
+
+**All PRs must include a commit message with the changes description!** 
+
+For the initial start, fork the project and use git clone command to download the repository to your computer. A standard procedure for working on an issue would be to:
```
Comment (christyjacob4) [inline CONTRIBUTING.md:30]:
```suggestion
For the initial start, fork the project and use the `git clone` command to download the repository to your computer. A standard procedure for working on an issue would be to:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+1. `git pull`, before creating a new branch, pull the changes from upstream. Your master needs to be up to date.
```
Comment (christyjacob4) [inline CONTRIBUTING.md:32]:
```suggestion
1. `git pull`, before creating a new branch, pull the changes from upstream. Your main needs to be up to date.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+```
+3. Work - commit - repeat ( be sure to be in your branch )
+
+4. Push changes to GitHub 
+```
+$ git push origin [name_of_your_new_branch]
+```
+
+5. Submit your changes for review
+If you go to your repository on GitHub, you'll see a `Compare & pull request` button. Click on that button.
```
Comment (christyjacob4) [inline CONTRIBUTING.md:48]:
```suggestion
5. Submit your changes for review. If you go to your repository on GitHub, you'll see a `Compare & pull request` button. Click on that button.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+### File Structure
+
+Below are outlined the most useful files for adding a new database adapter: 
+
+```bash
+.
+├── src # Source code
+│   └── Database
+│       ├── Adapter # Where your new adapter goes!
```
Comment (christyjacob4) [inline CONTRIBUTING.md:67]:
```suggestion
│       ├── Adapter/ # Where your new adapter goes!
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+│       ├── Adapter.php # Parent class for individual adapters
+│       ├── Database.php # Database class - calls individual adapter methods
+│       ├── Document.php # Document class - 
+│       └── Query.php # Query class - holds query attributes, methods, and values
+└── tests
+    └── Database
+        ├── Adapter # Extended from Base
```
Comment (christyjacob4) [inline CONTRIBUTING.md:74]:
```suggestion
        ├── Adapter/ # Extended from Base 
```
Same here

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate;
+use Utopia\Database\Query;
+use Utopia\Database\Validator\Authorization;
+
+class NewDB extends Adapter
+{
+```
+
+Only include dependencies strictly necessary for the database, preferably official PHP clients, if available.
```
Comment (christyjacob4) [inline CONTRIBUTING.md:100]:
```suggestion
Only include dependencies strictly necessary for the database, preferably official PHP libraries, if available.
```
Is libraries a better term?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+### Tips and Tricks
+
+- Keep it simple :)
+- Databases are namespaced, so name them with `$this->getNamespace()` from the parent Adapter class.
+- Create indexes for `$id`, `$read`, and `$write` fields by default for query performance.
+- Filter new IDs with `$this->filter($id);`
+- Prioritize code performance.
+- The Query and Queries validators contain the information about which queries the adapters support.
+- The [Authorization validator](/src/Database/Validator/Authorization.php) is used to consider permissions for searching methods `find()` and `sort()`. Only return documents with correct `read` permissions.
```
Comment (christyjacob4) [inline CONTRIBUTING.md:114]:
```suggestion
- The [Authorization validator](/src/Database/Validator/Authorization.php) is used to enforce permissions for searching methods `find()` and `sort()`. It only returns documents with the correct `read` permissions.
```

File: CONTRIBUTING.md (hunk 6 above)
Comment (kodumbeats) [reply CONTRIBUTING.md:100]: Works for me :)

**Review by christyjacob4 — APPROVED** [review]
Looks Good 👍

**Pull request merged**